modules/internal/ChapelHashtable.chpl

```diff
@@ -277,8 +265,7 @@ module ChapelHashtable {
       this.valType = valType;
       this.tableNumFullSlots = 0;
       this.tableNumDeletedSlots = 0;
-      this.tableSizeNum = 0;
-      this.tableSize = chpl__primes(tableSizeNum);
+      this.tableSize = 32; // start at 32 to avoid too small of table sizes
```

Comment:
Note that we explicitly made the starting size 0 in https://github.com/chapel-lang/chapel/pull/15918 to avoid paying for memory allocation for hashtables that are created but never used. That specific use-case where we had a regression isn't relevant anymore since we stopped using hashtables for tracking arrays declared over domains in https://github.com/chapel-lang/chapel/pull/16740, but I still wonder if it makes sense to start off empty and bump to 32 on first allocation.

Reply:
That would make sense to me; the reason I have it set to 32 now is there were some failing tests that passed changing this to 32, but that just indicates that there is still something wrong more than anything. I suspect the failures are coming due to the tests explicitly expecting the old probing method and such (they are calling the `ChapelHashtable` internal functions and testing they match hard coded values) and not an issue with these changes, but haven't been able to get it all the way figured out yet. 

Either way, I agree it should start at 0 and then jump to something bigger like 32 or whatever people think makes the most sense on the first resize. Right now I have it jumping to 2 in the case of a table size of 0, which of course wouldn't be the most useful map.